src/poetry/publishing/uploader.py

```diff
@@ -275,6 +319,12 @@ def _upload_file(
                 elif resp.status_code == 400 and "was ever registered" in resp.text:
                     self._register(session, url)
                     resp.raise_for_status()
+                elif _ignore_existing(resp, skip_existing):
+                    bar.set_format(
+                        f" - Uploading <c1>{file.name}</c1> <warning>File exists."
+                        " Skipping</>"
+                    )
+                    bar.finish()
```

Comment:
i think i'm slightly confused by the (even undecorated) output:

```
$ poetry publish -r localdev --skip-existing >out 2>err
$ cat out

Publishing poetry-skip (0.1.0) to localdev
 - Uploading poetry-skip-0.1.0.tar.gz File exists. Skipping
 - Uploading poetry_skip-0.1.0-py3-none-any.whl File exists. Skipping
$ cat err
 - Uploading poetry-skip-0.1.0.tar.gz 0%
 - Uploading poetry-skip-0.1.0.tar.gz 100% - Uploading poetry_skip-0.1.0-py3-none-any.whl 0%
 - Uploading poetry_skip-0.1.0-py3-none-any.whl 100%
```

(`out` seems ok to me even without the percentages and separate lines that you list above)

Reply:
Hmm, for undecorated IO there is an early return in `finish()` because `self._should_overwrite` is `False` (https://github.com/sdispater/cleo/blob/7ebf50b4c85e0eca2d9f295588150527c95c84de/cleo/ui/progress_bar.py#L223-L224). I can't reproduce the `File exists. Skipping` messages in out. Actually, `ProgressBar` should do only error out (https://github.com/sdispater/cleo/blob/7ebf50b4c85e0eca2d9f295588150527c95c84de/cleo/ui/progress_bar.py#L50-L52).

Are the `File exists. Skipping` messages in the output if you run the `test_publish_dry_run_skip_existing` test case?

The missing newline between 100% and the next 0% seems to be an existing issue that has nothing to do with `--skip-existing`. Thus, we can ignore that. But the `File exists. Skipping` should be in error output.

Reply:
sorry i forgot to mention: your initial comment was correct in that the output was broken. i fixed it (`bar.set_format` -> `self._io.overwrite`) and the above was me trying to test my fix

sidenote: do you mind me force-pushing (gh still shows such diffs) or would you prefer fixup commits that we squash once all comments are addressed?